tests/integration_tests/functional/test_cpu_features_aarch64.py

```diff
@@ -27,37 +28,57 @@
+)
+
+DEFAULT_G3_FEATURES_NO_SSBS_5_10 = (
+    "fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp "
+    "asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 sm3 sm4 asimddp "
+    "sha512 asimdfhm dit uscat ilrcpc flagm dcpodp i8mm bf16 dgh rng"
+)
+
+DEFAULT_G3_FEATURES_WITH_SVE_AND_PAC_5_10 = (
+    "fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp "
+    "asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 sm3 sm4 asimddp "
+    "sha512 sve asimdfhm dit uscat ilrcpc flagm ssbs paca pacg dcpodp svei8mm svebf16 i8mm bf16 dgh rng"
+)
+
 DEFAULT_G3_FEATURES_V1N1 = (
     "fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp "
     "asimdhp cpuid asimdrdm lrcpc dcpop asimddp ssbs"
 )
 
 
-def _check_cpu_features_arm(test_microvm, template_name=None):
+def _check_cpu_features_arm(test_microvm, kernel, template_name=None):
     expected_cpu_features = {"Flags": []}
-    match cpuid_utils.get_instance_type():
-        case "m6g.metal":
-            match template_name:
-                case "aarch64_remove_ssbs":
-                    expected_cpu_features["Flags"] = DEFAULT_G2_FEATURES_NO_SSBS
-                case "aarch64_v1n1":
-                    expected_cpu_features["Flags"] = DEFAULT_G2_FEATURES
-                case None:
-                    expected_cpu_features["Flags"] = DEFAULT_G2_FEATURES
-        case "c7g.metal":
-            match template_name:
-                case "aarch64_remove_ssbs":
-                    expected_cpu_features["Flags"] = DEFAULT_G3_FEATURES_NO_SSBS
-                case "aarch64_v1n1":
-                    expected_cpu_features["Flags"] = DEFAULT_G3_FEATURES_V1N1
-                case None:
-                    expected_cpu_features["Flags"] = DEFAULT_G3_FEATURES
+    match (cpuid_utils.get_instance_type(), kernel, template_name):
+        case ("m6g.metal", _, "aarch64_remove_ssbs"):
+            expected_cpu_features["Flags"] = DEFAULT_G2_FEATURES_NO_SSBS
+        case ("m6g.metal", _, "aarch64_v1n1"):
+            expected_cpu_features["Flags"] = DEFAULT_G2_FEATURES
+        case ("m6g.metal", _, None):
+            expected_cpu_features["Flags"] = DEFAULT_G2_FEATURES
+        case ("c7g.metal", "4.14", "aarch64_remove_ssbs"):
+            expected_cpu_features["Flags"] = DEFAULT_G3_FEATURES_NO_SSBS_4_14
+        case ("c7g.metal", "5.10", "aarch64_remove_ssbs"):
+            expected_cpu_features["Flags"] = DEFAULT_G3_FEATURES_NO_SSBS_5_10
+        case ("c7g.metal", "4.14", "aarch64_with_sve_and_pac"):
+            expected_cpu_features["Flags"] = DEFAULT_G3_FEATURES_WITH_SVE_AND_PAC_4_14
+        case ("c7g.metal", "5.10", "aarch64_with_sve_and_pac"):
+            expected_cpu_features["Flags"] = DEFAULT_G3_FEATURES_WITH_SVE_AND_PAC_5_10
+        case ("c7g.metal", _, "aarch64_v1n1"):
+            expected_cpu_features["Flags"] = DEFAULT_G3_FEATURES_V1N1
+        case ("c7g.metal", _, None):
+            expected_cpu_features["Flags"] = DEFAULT_G3_FEATURES
 
```

Comment:
This looks like it could be a dict. Also it may be simpler to manipulate the list of features as a set rather than as an opaque string:

```python
In [2]: {"fp", "asimd", "ssbs"} - {"ssbs"}
Out[2]: {'asimd', 'fp'}
```

Reply:
It is a string, because `cpuid_utils.check_guest_cpuid_output` is expecting a string.

Reply:
Yeah I understand that is a tangential refactor, but it would look a bit better than trying to currently do the same with strings.

```python
set(x for x in stdout.split(" "))
```